lint/unexported-call_checker.go

```diff
@@ -0,0 +1,79 @@
+package lint
+
+import (
+	"fmt"
+	"go/ast"
+	"go/types"
+)
+
+type unexportedCallChecker struct {
+	ctx *Context
+}
+
+func newUnexportedCallChecker(ctx *Context) Checker {
+	return &unexportedCallChecker{
+		ctx: ctx,
+	}
+}
+
+// Check finds calls of unexported method from unexported type
+// outside that type.
+//
+//TODO: update description and warning message
+func (c *unexportedCallChecker) Check(f *ast.File) {
+	for _, decl := range collectFuncDecls(f) {
+		if decl.Body == nil {
+			continue
+		}
+		name := ""
+		cond := decl.Recv != nil &&
+			decl.Recv.List != nil &&
+			len(decl.Recv.List) == 1 &&
+			decl.Recv.List[0].Names != nil &&
+			len(decl.Recv.List[0].Names) == 1
```

Comment:
I believe two nil checks for `List` fields are redundant as nil slice length is 0.
The `== 1` check would cover both cases and is perfectly safe.

Reply:
The `gosimple` doesn't do a best job with it's [S1009](https://staticcheck.io/docs/gosimple#S1009).
It should catch these cases as well.